felix/bpf/counters/counters.go

```diff
@@ -164,7 +164,11 @@ func (c Counters) Read() ([][]uint64, error) {
 	for hook, name := range bpf.Hooks {
 		val, err := c.read(c.maps[hook])
 		if err != nil {
-			return values, fmt.Errorf("Failed to read bpf counters. hook=%s err=%v", name, err)
+			if name == bpf.HookXDP {
+				logrus.Infof("Failed to read XDP bpf counters. err=%v", err)
+				continue
+			}
+			return values, fmt.Errorf("Failed to read bpf counters. hook=%s err=%w", name, err)
 		}
 		if len(values[hook]) < MaxCounterNumber {
```

Comment:
Is this ever possible given that on line 161 you allocate exactly this much space?

I think it is meant to test `len(val)`  and not to allocate at 161